src/components/new-safe/steps/Step2/index.tsx

```diff
@@ -49,11 +49,18 @@ const CreateSafeStep2 = ({
     },
   })
 
-  const { register, handleSubmit, control, watch, formState, getValues } = formMethods
+  const { handleSubmit, control, watch, formState, getValues, setValue } = formMethods
 
   const threshold = watch(CreateSafeStep2Fields.threshold)
 
-  const { fields: ownerFields, append: appendOwner, remove: removeOwner } = useFieldArray({ control, name: 'owners' })
+  const { fields: ownerFields, append: appendOwner, remove } = useFieldArray({ control, name: 'owners' })
+
+  const removeOwner = (index: number): void => {
+    if (ownerFields.length === threshold) {
```

Comment:
I feel like `setValue('threshold', Math.min(threshold, ownerField.length - 1))` is more direct.
But this might be very subjective.